engine/gamesys/src/gamesys/components/comp_sprite.cpp

```diff
@@ -236,16 +239,20 @@ namespace dmGameSystem
         return result;
     }
 
-    static inline dmRender::HMaterial GetMaterial(const SpriteComponent* component, const SpriteResource* resource) {
+    static inline MaterialResource* GetMaterialResource(const SpriteComponent* component, const SpriteResource* resource) {
         return component->m_Material ? component->m_Material : resource->m_Material;
     }
 
-    static inline TextureSetResource* GetTextureSet(const SpriteComponent* component, const SpriteResource* resource) {
-        return component->m_TextureSet ? component->m_TextureSet : resource->m_TextureSet;
+    static inline dmRender::HMaterial GetMaterial(const SpriteComponent* component, const SpriteResource* resource) {
+        return GetMaterialResource(component, resource)->m_Material;
+    }
+
+    static inline TextureSetResource* GetTextureSet(const SpriteComponent* component) {
+        return component->m_TextureSet ? component->m_TextureSet : component->m_Resource->m_TextureSet;
     }
```

Comment:
A lot of these functions are duplicated, or at the very least very similar, for most components. Would it make sense to move these to a header file to remove a bit of code in the components?